include/xsparse/level_capabilities/co_iteration.hpp

```diff
@@ -0,0 +1,202 @@
+#ifndef XSPARSE_CO_ITERATION_HPP
+#define XSPARSE_CO_ITERATION_HPP
+
+#include <vector>
+#include <tuple>
+#include <algorithm>
+#include <climits>
+
+namespace xsparse::level_capabilities
+{
+    template <class F, class IK, class PK, class Levels, class Is>
+    class Coiterate;
+
+    template <class F, class IK, class PK, class... Levels, class... Is>
+    class Coiterate<F, IK, PK, std::tuple<Levels...>, std::tuple<Is...>>
+    {
+    public:
+        class coiteration_helper
+        {
+        public:
+            std::tuple<Is...> const m_i;
+            PK const m_pkm1;
+            std::tuple<Levels&...> const m_levelsTuple;
+            F const m_comparisonHelper;
+
+        public:
+            explicit inline coiteration_helper(F f,
+                                               std::tuple<Is...>& i,
+                                               PK& pkm1,
+                                               Levels&... levels) noexcept
+                : m_i(std::move(i))
+                , m_pkm1(std::move(pkm1))
+                , m_levelsTuple(std::tie(levels...))
+                , m_comparisonHelper(f)
+            {
+                static_assert(std::tuple_size_v<decltype(m_levelsTuple)> >= 2,
+                              "Tuple size should be at least 2");
+            }
+
+            class iterator
+            {
+            private:
+                coiteration_helper const& m_coiterHelper;
+                std::tuple<typename Levels::LevelCapabilities::iteration_helper::iterator...>
+                    iterators;
+                IK min_ik;
+
+            private:
+                template <typename... T1, typename... T2>
+                inline constexpr auto compareHelper(const std::tuple<T1...>& t1,
+                                                    const std::tuple<T2...>& t2)
+                {
+                    static_assert(sizeof...(T1) == sizeof...(T2));
```

Comment:
This should be reversed, and you should calculate `!=` this way, and reverse that to get `==`.

Reply:
Yeah, and also the operators need to be `const` so that was an issue as well.

Reply:
This still needs addressing.

Reply:
This still needs addressing.

Applying `!=` to the individual iterators and passing it through `f` gives us the result of the `!=` operator.

Flipping that gives us the result of `==`.

The logic needs to be flipped here.